### PR TITLE
Fix Function Controller manifests generation

### PR DESCRIPTION
**Pull request by tgorgol**
**Description**

Changes proposed in this pull request:

- Fix Function Controller manifests generation

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/11362

### CODE REVIEW

File: components/function-controller/Makefile [hunk 1]
```diff
@@ -31,20 +38,36 @@ test-local:
...
+# download controller-gen if necessary
+.PHONY: controller-gen-local
+controller-gen-local:
+ifeq (, $(shell which controller-gen))
+	@{ \
+	set -e ;\
+	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
+	cd $$CONTROLLER_GEN_TMP_DIR ;\
+	go mod init tmp ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.9 ;\
```
Comment (pPrecel) [inline components/function-controller/Makefile:63]: Are there any breaking changes between version 0.2.9 and 0.2.4? I found in our `README.md` that we require `controller-gen (v0.2.4+)`. Maybe can we change this to `controller-gen (v0.2.9)`? Or is this not necessary?

**Review by pPrecel — APPROVED** [review]
LGTM

File: components/function-controller/Makefile (hunk 1 above)
Comment (tgorgol) [reply components/function-controller/Makefile:63]: I can update the readme with the new version but looking at the [changelog of controller-tools](https://github.com/kubernetes-sigs/controller-tools/releases) I don't see any breaking changes from `0.2.4` to `0.2.9`. Used this version as it was the version our current CRDs were manually generated with in recent updates related to Kyma 2.0.

**Review by pPrecel — APPROVED** [review]

**Pull request merged**